platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ExportMigrationManagerImpl.java

```diff
@@ -174,18 +176,22 @@ public void doExport(String productBranding, String productVersion) {
   public void close() throws IOException {
     if (!closed) {
       this.closed = true;
-      CipherOutputStream cos = cipherUtils.getCipherOutputStream(zipOutputStream);
+      CipherOutputStream cos = null;
       try {
         zipOutputStream.closeEntry();
         try {
           zipOutputStream.putNextEntry(
               new ZipEntry(MigrationContextImpl.METADATA_FILENAME.toString()));
+          CloseShieldOutputStream shieldOutputStream = new CloseShieldOutputStream(zipOutputStream);
+          cos = cipherUtils.getCipherOutputStream(shieldOutputStream);
           JsonUtils.MAPPER.writeValue(cos, metadata);
         } catch (IOException e) {
           throw new MigrationException(Messages.EXPORT_METADATA_CREATE_ERROR, e);
         }
       } finally {
-        cos.close();
+        if (cos != null) {
+          cos.close();
```

Comment:
❗️ Since this close might throw an exception, what will happen with the zipOutputStream one? Might want to close it in a finally close. Or if I may suggest, move this close into a finally for the previous try/catch. That way this one will always be called to close the zip output stream.